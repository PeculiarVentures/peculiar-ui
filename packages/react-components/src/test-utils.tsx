import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
