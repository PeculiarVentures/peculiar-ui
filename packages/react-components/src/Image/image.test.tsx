import React from 'react';
import { renderWithWrapper as render, screen, fireEvent } from '../test-utils';
import { Image } from '../index';

const src = 'https://images.unsplash.com/photo-1667489022797-ab608913feeb?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHw5fHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60';

describe('<Image />', () => {
  it('should render as default', () => {
    const { asFragment } = render(<Image src={src} />);

    const img = screen.getByRole('img');

    expect(img).toBeInTheDocument();
    expect(img.tagName).toBe('IMG');
    expect(img.getAttribute('src')).toEqual(src);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have test id', () => {
    const { asFragment } = render(<Image src={src} data-testid="test-id" />);

    const img = screen.getByRole('img');

    expect(img.getAttribute('data-testid')).toMatch(/test-id/i);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have alt', () => {
    const { asFragment } = render(<Image src={src} alt="Test alt" />);

    const img = screen.getByRole('img');

    expect(img.getAttribute('alt')).toMatch(/Test alt/i);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have class name', () => {
    const { asFragment } = render(<Image src={src} className="test-cls" />);

    const img = screen.getByRole('img');

    expect(img.getAttribute('class')).toMatch(/test-cls/i);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onError handler', () => {
    const handleErr = jest.fn();

    render(<Image src="broken.png" onError={handleErr} />);

    const img = screen.getByRole('img');

    fireEvent.error(img);

    expect(handleErr).toHaveBeenCalledTimes(1);
  });

  it('should call onLoad handler', () => {
    const handleLoad = jest.fn();

    render(<Image src={src} onLoad={handleLoad} />);

    const img = screen.getByRole('img');

    fireEvent.load(img);

    expect(handleLoad).toHaveBeenCalledTimes(1);
  });

  it('should render loading component', () => {
    const { asFragment } = render(<Image src={undefined} loading={<div>Loading</div>} />);

    expect(screen.getByText(/Loading/i)).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
