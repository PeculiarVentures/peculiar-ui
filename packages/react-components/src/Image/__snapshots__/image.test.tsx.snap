// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should have alt 1`] = `
<DocumentFragment>
  <img
    alt="Test alt"
    src="https://images.unsplash.com/photo-1667489022797-ab608913feeb?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHw5fHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60"
  />
</DocumentFragment>
`;

exports[`<Image /> should have class name 1`] = `
<DocumentFragment>
  <img
    class="test-cls"
    src="https://images.unsplash.com/photo-1667489022797-ab608913feeb?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHw5fHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60"
  />
</DocumentFragment>
`;

exports[`<Image /> should have test id 1`] = `
<DocumentFragment>
  <img
    data-testid="test-id"
    src="https://images.unsplash.com/photo-1667489022797-ab608913feeb?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHw5fHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60"
  />
</DocumentFragment>
`;

exports[`<Image /> should render as default 1`] = `
<DocumentFragment>
  <img
    src="https://images.unsplash.com/photo-1667489022797-ab608913feeb?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHw5fHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60"
  />
</DocumentFragment>
`;

exports[`<Image /> should render loading component 1`] = `
<DocumentFragment>
  <div>
    Loading
  </div>
</DocumentFragment>
`;
